chia/full_node/full_node.py

```diff
@@ -709,101 +709,100 @@ async def _sync(self):
-                        peak = self.blockchain.get_peak()
-                        if peak is not None and block_response.block.prev_header_hash == peak.header_hash:
-                            fork_point_height = our_peak_height
-                        break
-
-        for i in range(fork_point_height, target_peak_sb_height, batch_size):
-            start_height = i
-            end_height = min(target_peak_sb_height, start_height + batch_size)
-            request = RequestBlocks(uint32(start_height), uint32(end_height), True)
-            self.log.info(f"Requesting blocks: {start_height} to {end_height}")
-            batch_added = False
-            to_remove = []
-            for peer in peers_with_peak:
-                if peer.closed:
-                    to_remove.append(peer)
-                    continue
-                response = await peer.request_blocks(request, timeout=60)
-                if response is None:
-                    await peer.close()
-                    to_remove.append(peer)
-                    continue
-                if isinstance(response, RejectBlocks):
-                    to_remove.append(peer)
-                    continue
-                elif isinstance(response, RespondBlocks):
-                    success, advanced_peak, _ = await self.receive_block_batch(
-                        response.blocks, peer, None if advanced_peak else uint32(fork_point_height), summaries
-                    )
-                    if success is False:
-                        await peer.close(600)
-                        continue
-                    else:
-                        batch_added = True
-                        break
-
-            peak = self.blockchain.get_peak()
-            assert peak is not None
-            msg = make_msg(
-                ProtocolMessageTypes.new_peak_wallet,
-                wallet_protocol.NewPeakWallet(
-                    peak.header_hash,
-                    peak.height,
-                    peak.weight,
-                    uint32(max(peak.height - 1, uint32(0))),
-                ),
-            )
-            await self.server.send_to_all([msg], NodeType.WALLET)
+        async def fetch_block_batches(batch_queue, peers_with_peak: List):
+            try:
+                for start_height in range(fork_point_height, target_peak_sb_height, batch_size):
+                    end_height = min(target_peak_sb_height, start_height + batch_size)
+                    request = RequestBlocks(uint32(start_height), uint32(end_height), True)
+                    fetched = False
+                    for peer in random.sample(peers_with_peak, len(peers_with_peak)):
+                        if peer.closed:
+                            peers_with_peak.remove(peer)
+                            continue
+                        response = await peer.request_blocks(request, timeout=10)
```

Comment:
can this throw? suppose the peer times-out or gets disconnected for some reason. If that throws, it would make sense to catch that here and just treat this peer as ineligible

Reply:
not sure but we do have a try surrounding this now